eclass/rocm.eclass

```diff
@@ -176,9 +176,12 @@ _rocm_set_globals() {
 	local optflags=${allflags[@]/%/(-)?}
 	ROCM_USEDEP=${optflags// /,}
 }
-_rocm_set_globals
-unset -f _rocm_set_globals
 
+# dev-util/hip uses checkgpu but does not need the USE flags.
+if [[ ${CATEGORY}/${PN} != dev-util/hip ]]; then
```

Comment:
Something like, and insert a `ROCM_ECLASS_NO_GLOBALS=true` before `inherit rocm.eclass` in `dev-util/hip-5.7.1-r1.ebuild`

```suggestion
if [[ -n ${ROCM_ECLASS_NO_GLOBALS} ]]; then
```